### PR TITLE
tests: Add fuzzing harness for CheckTransaction(...), IsStandardTx(...) and other CTransaction related functions

### PR DESCRIPTION
**Pull request by practicalswift**
Add fuzzing harness for `CheckTransaction(...)`, `IsStandardTx(...)` and other `CTransaction` related functions.

**Testing this PR**

Run:

```
$ CC=clang CXX=clang++ ./configure --enable-fuzz --with-sanitizers=address,fuzzer,undefined
$ make
$ src/test/fuzz/transaction
…
# And to to quickly verify that the relevant code regions are triggered, that the
# fuzzing throughput seems reasonable, etc.
$ contrib/devtools/test_fuzzing_harnesses.sh '^transaction$'
```

`test_fuzzing_harnesses.sh` can be found in PR #17000.

### CODE REVIEW

File: src/test/fuzz/transaction.cpp [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        ds >> nVersion;
+        ds.SetVersion(nVersion);
+    } catch (const std::ios_base::failure& e) {
+        return;
+    }
+    const CTransaction tx = [&] {
+        try {
+            return CTransaction(deserialize, ds);
+        } catch (const std::ios_base::failure& e) {
+            return CTransaction();
```
Comment (MarcoFalke) [inline src/test/fuzz/transaction.cpp]: why not return early here. Seems wasteful to map all deserialization failures to an default constructed tx and then run all the checks with that.

Also, `TRANSACTION_DESERIALIZE` can be removed from the existing fuzzer now, as the new fuzz test includes this dumb code path now?

Comment (practicalswift) [reply src/test/fuzz/transaction.cpp]: Good points. Feedback addressed. Please re-review :)

File: src/Makefile.test.include [hunk 2]
```diff
@@ -321,6 +315,12 @@ test_fuzz_blocktransactionsrequest_deserialize_CPPFLAGS = $(AM_CPPFLAGS) $(BITCO
 test_fuzz_blocktransactionsrequest_deserialize_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 test_fuzz_blocktransactionsrequest_deserialize_LDFLAGS = $(RELDFLAGS) $(AM_LDFLAGS) $(LIBTOOL_APP_LDFLAGS)
 test_fuzz_blocktransactionsrequest_deserialize_LDADD = $(FUZZ_SUITE_LD_COMMON)
+test_fuzz_transaction_SOURCES = $(FUZZ_SUITE) test/fuzz/transaction.cpp
```
Comment (MarcoFalke) [inline src/Makefile.test.include]: style-nit: new line?

File: test/fuzz/test_runner.py [hunk 3]
```diff
@@ -12,6 +12,10 @@
 import subprocess
 import logging
 
+# Fuzzers known to lack a seed corpus in https://github.com/bitcoin-core/qa-assets/tree/master/fuzz_seed_corpus
+FUZZERS_MISSING_CORPORA = [
+    "transaction",
```
[the remark this answers is not in the document]
Comment (practicalswift) [reply test/fuzz/test_runner.py]: Fixed!

File: src/Makefile.test.include (hunk 2 above)
Comment (practicalswift) [reply src/Makefile.test.include]: Fixed!

**Comment by MarcoFalke** [comment]
ACK 5c2987636faa5bc175b37b81fd98ab48e576da0b

**Pull request merged**